cylc/flow/task_pool.py

```diff
@@ -229,7 +233,10 @@ def add_to_runahead_pool(self, itask, is_new=True):
         self.runahead_pool[itask.point][itask.identity] = itask
         self.rhpool_changed = True
 
-        # add row to "task_states" table
+        # Register pool node in data-store in external ID format
+        self.data_store_mgr.add_runahead_node(itask.tdef.name, itask.point)
```

Comment:
What's "external ID format"?  (It's not mentioned anywhere else in cylc-flow).

Why are runahead tasks added here, rather than just active tasks?

Reply:
> Why are runahead tasks added here, rather than just active tasks?

I use it to bridge the gap between removed parent and active child.